metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_connection.py

```diff
@@ -110,6 +109,10 @@ class SnowflakeConnectionConfig(ConfigModel):
         default=None,
         description="OAuth token from external identity provider. Not recommended for most use cases because it will not be able to refresh once expired.",
     )
+    snowflake_domain: str = pydantic.Field(
+        default=DEFAULT_SNOWFLAKE_DOMAIN,
+        description="Snowflake domain. Use 'snowflakecomputing.com' for most regions or 'snowflakecomputing.cn' for China (cn-northwest-1) region.",
```

Comment:
would it be possible for us to automatically figure this out from the region / account ID?

Reply:
I considered that but there are some legacy account ids without the region info. While we could likely overcome that, I think that:
- as of today, default works for most 99.99999% of users
- explicit is better than implicit, this can be opinionated though
- future-proof if snowflake adds more regions with different domains